File: tests/fixtures/python/t101_violation.py
def test_user_creation_calls_repository(mock_repo):
    service = UserService(mock_repo)
    service.create_user("alice")
    mock_repo.save.assert_called_with("alice")
    mock_repo.save.assert_called_once()
    assert service is not None


def test_notification_sent(mock_notifier):
    service = OrderService(mock_notifier)
    service.place_order(item="book")
    mock_notifier.send.assert_called_once_with(item="book")
    assert True

File: tests/fixtures/php/t101_violation.php
<?php

class UserServiceTest extends \PHPUnit\Framework\TestCase
{
    public function test_create_user_calls_repository(): void
    {
        $repo = $this->createMock(UserRepository::class);
        $repo->expects($this->once())
            ->method('save');

        $service = new UserService($repo);
        $service->createUser("alice");
    }

    public function test_notify_uses_mockery(): void
    {
        $notifier = \Mockery::mock(Notifier::class);
        $notifier->shouldReceive('send')
            ->once()
            ->with('hello');

        $service = new UserService($notifier);
        $service->notify("hello");
    }
}
